packages/apm/src/span.ts

```diff
@@ -324,10 +283,21 @@ export class Span implements SpanInterface, SpanContext {
       logger.warn('Discarding transaction Span without sampling decision');
       return undefined;
     }
+
     const finishedSpans = this.spanRecorder ? this.spanRecorder.finishedSpans.filter(s => s !== this) : [];
 
+    // The reason we do `useLastSpanTimestamp` is that if we use the Tracing integration with an idle transaction.
+    // The idea is that after e.g. 500ms idle time, we finish the transaction
+    // But we don't want the "idle time" taking into account for the end timestamp of the transaction
+    // instead we search for the timestamp of the last finished span and change the end timestamp to the same
+    // timestamp of the last finished span
     if (useLastSpanTimestamp && finishedSpans.length > 0) {
-      this.timestamp = finishedSpans[finishedSpans.length - 1].timestamp;
+      this.timestamp = finishedSpans.reduce((prev: Span, current: Span) => {
+        if (prev.timestamp && current.timestamp) {
+          return prev.timestamp > current.timestamp ? prev : current;
+        }
+        return prev;
+      }).timestamp;
```

Comment:
```js
> [undefined, 1].reduce((prev, cur) => {
    if (prev && cur) {
        return prev > cur ? prev : cur;
    }
    return prev;
})
undefined
```

@HazAT can `finishedSpans` contain spans where `timestamp` is not set? Why do we even check `if (prev.timestamp && current.timestamp)`?

Reply:
In case there is only 1 span in finishedSpans, this code does the equivalent of:

```js
this.timestamp = finishedSpans[0].timestamp
```

If `finishedSpans[i].timestamp` is allowed to be null/undefined, then we'd be potentially clearing the end timestamp instead of the intended behavior of trimming towards the end.

Reply:
so the property `timestamp` is defined like:`public timestamp?: number;` that makes it necessary to always null check it. That's the only reason.
By definition a finishedSpan is a Span with timestamp set, so this is :fine:

Reply:
Alright